aikit/transformers/categories.py

```diff
@@ -477,3 +479,231 @@ def _get_model(self, X, y=None):
 
 
 # In[]
+
+class _OrdinalOneHotEncoder(BaseEstimator, TransformerMixin):
+
+    def __init__(self,
+                 categories="auto",
+                 dtype=np.int32
+                 ):
+        self.categories=categories
+        self.dtype=dtype
+        
+        
+    def get_feature_names(self, input_features=None):
+        
+        if input_features is None:
+            input_features = self._input_features
+            
+        if len(input_features) != self._nb_columns:
+            raise ValueError("input_features (length = (%d)) should have the same length has the number of columns (%d)" % (len(input_features), self._nb_columns))
+
+        features = []
+        for j, f in enumerate(input_features):
+            features += [str(f) + "_g_" + str(m) for m in self._all_inv_mapping[j].values[:-1]]
+            
+        return features
+
+    def fit(self, X, y=None):
+        
+        if get_type(X) != DataTypes.DataFrame:
+            raise TypeError("This transformer only works for DataFrame")
```

Comment:
Why should it only work on dataframe?

Reply:
because it is a mess to make it work for all the data-format ... and so i force a DataFrame.
Here it is in the hidden class : the wrapped class accept everything.